qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,623 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name,redefined-builtin,method-hidden,arguments-differ
+
+"""Module providing definitions of Pulse Qobj classes."""
+
+import copy
+import json
+
+import numpy
+
+from qiskit.qobj.qasm_qobj import QobjHeader
+from qiskit.qobj.qasm_qobj import QasmQobjExperimentHeader
+from qiskit.qobj.qasm_qobj import validator
+
+
+class QobjMeasurementOption:
+    """An individual measurement option."""
+
+    def __init__(self, name, params=None):
+        """Instantiate a new QobjMeasurementOption object.
+
+        Args:
+            name (str): The name of the measurement option
+            params (list): The parameters of the measurement option.
+        """
+        self._data = {}
+        self._data['name'] = name
```

Comment:
The main reason I did this was for the ease of implementing the `to_dict()` method for classes that accept arbitrary dict input (headers, config, etc). To avoid extra unused parameters I tried to stick to attributes for the classes which don't accept arbitrary fields (also typically a straight attribute will be slightly faster for then a dict access from an attribute). That being said I probably wasn't entirely consistent on this aspect because of the size of the change I likely got sloppy and copy and pasted the wrong pattern for some of the classes. I will double check it all and update that on my next revision

As for using `SimpleNamespace` that's a really good idea, I forgot that exists (sometimes I'm still too used to python 2). I'll look into switching the classes which use a private dict to that and see how simple it is.

Reply:
So I just double checked and at least on the current state of the PR all the PRs that use `_data` a like this class all take arbitrary key value pairs, except for this example. I will update  `QobjMeasurementOption` to use attrs instead of a dict. I still need to investigate the usage of simple namespace to see which is easier to work with and do some profiling and benchmarking to see if there is an advantage in performance either way.

Reply:
I updated this in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/69e084b3dbda026e48493eeaaa497666c8278e96

Reply:
Pivotted to SimpleNamespace for the arbitrary KV pair classes in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/14eb5180f8b15bddde0b20ae2701a1f6c0b49dd7